pubnub-gson/src/main/java/com/pubnub/internal/v2/subscription/SubscriptionImpl.kt

```diff
@@ -26,7 +28,11 @@ class SubscriptionImpl(
     channels: Set<ChannelName>,
     channelGroups: Set<ChannelGroupName>,
     options: SubscriptionOptions,
-) : Subscription, BaseSubscriptionImpl<EventListener>(pubnub.pubNubCore, channels, channelGroups, options) {
+    eventEmitterFactory: (BaseSubscriptionImpl<EventListener>) -> EventEmitterImpl = { baseSubscriptionImpl ->
+        EventEmitterImpl(AnnouncementCallback.Phase.SUBSCRIPTION, baseSubscriptionImpl::accepts)
+    },
+) : Subscription,
+    BaseSubscriptionImpl<EventListener>(pubnub.pubNubCore, channels, channelGroups, options, eventEmitterFactory) {
     // todo maybe we can add it to constructor so we can mock it?
```

Comment:
would it be worth doing it in the same PR to check if it's possible now, with the new changes? unless you've checked and you know it's possible without making further changes to the eventEmitter

Reply:
I just saw your description "For now I don't have simple solution how to move emitterHelper to constructor"

let me take a quick look